pkg/epp/scheduling/framework/plugins/filter/least_kvcache_filter.go

```diff
@@ -35,26 +35,43 @@ var _ framework.Filter = &LeastKVCacheFilter{}
 
 // LeastKVCacheFilterFactory defines the factory function for LeastKVCacheFilter.
 func LeastKVCacheFilterFactory(name string, _ json.RawMessage, _ plugins.Handle) (plugins.Plugin, error) {
-	return NewLeastKVCacheFilter(), nil
+	return NewLeastKVCacheFilter().WithName(name), nil
 }
 
 // NewLeastKVCacheFilter initializes a new LeastKVCacheFilter and returns its pointer.
 func NewLeastKVCacheFilter() *LeastKVCacheFilter {
-	return &LeastKVCacheFilter{}
+	f := &LeastKVCacheFilter{}
+	return f.WithName("")
```

Comment:
instead of passing an empty string and then check it in the `WithName` function, you can do the following:
```suggestion
	return &LeastKVCacheFilter{
		name: LeastKVCacheFilterType
	}
```

if `WithName` will be called, it will override the name cause it's called always after the New function (and then no need to check in that function).